pkgs/applications/misc/nut/default.nix

```diff
@@ -8,42 +8,46 @@ stdenv.mkDerivation rec {
 
   src = fetchurl {
     url = "https://networkupstools.org/source/${lib.versions.majorMinor version}/${pname}-${version}.tar.gz";
-    sha256 = "19r5dm07sfz495ckcgbfy0pasx0zy3faa0q7bih69lsjij8q43lq";
+    sha256 = "sha256-w+WnCNp5e3xwtlPTexIGoAD8tQO4VRn+TN9jU/eSv+U=";
   };
 
   patches = [
-    (fetchpatch {
-      # Fix build with openssl >= 1.1.0
-      url = "https://github.com/networkupstools/nut/commit/612c05efb3c3b243da603a3a050993281888b6e3.patch";
-      sha256 = "0jdbii1z5sqyv24286j5px65j7b3gp8zk3ahbph83pig6g46m3hs";
-    })
     (substituteAll {
       src = ./hardcode-paths.patch;
       avahi = "${avahi}/lib";
       freeipmi = "${freeipmi}/lib";
-      libusb = "${libusb-compat-0_1}/lib";
+      libusb = "${libusb1}/lib";
       neon = "${neon}/lib";
+      libmodbus = "${libmodbus}/lib";
+      netsnmp = "${net-snmp.lib}/lib";
     })
   ];
 
-  buildInputs = [ neon libusb-compat-0_1 openssl udev avahi freeipmi ];
+  buildInputs = [ neon libusb1 openssl udev avahi freeipmi libmodbus i2c-tools net-snmp gd ];
 
   nativeBuildInputs = [ autoreconfHook libtool pkg-config makeWrapper ];
 
   configureFlags =
     [ "--with-all"
       "--with-ssl"
-      "--without-snmp" # Until we have it ...
       "--without-powerman" # Until we have it ...
-      "--without-cgi"
-      "--without-hal"
-      "--with-systemdsystemunitdir=$(out)/etc/systemd/system"
+      "--with-systemdsystemunitdir=$(out)/lib/systemd/system"
+      "--with-systemdshutdowndir=$(out)/lib/systemd/system-shutdown"
+      "--with-systemdtmpfilesdir=$(out)/lib/tmpfiles.d"
       "--with-udev-dir=$(out)/etc/udev"
     ];
 
+  # TODO: NUT installs systemd services, but these are largely useless, because they assume that configuration files
+  # exist $out/etc. It's probably the job of a NixOS module like power.ups to provide usable services. Does that mean we
+  # should delete the services that NUT installs?
+
   enableParallelBuilding = true;
 
-  env.NIX_CFLAGS_COMPILE = toString [ "-std=c++14" ];
+  postInstall = ''
+    substituteInPlace $out/lib/systemd/system-shutdown/nutshutdown \
+      --replace /bin/sleep "/run/current-system/systemd/bin/sleep" \
```

Comment:
This will make it work exclusively on NixOS (which covers probably 99% of use cases? fine for me), but make it unusable on a non-NixOS system. 
A more general approach could be to to just call `sleep` and set an appropriate path within the service definition.

Reply:
```suggestion
      --replace /bin/sleep "${coreutils}/bin/sleep" \
```